source/grid/grid_out.cc

```diff
@@ -1834,8 +1835,11 @@ GridOut::write_svg(const Triangulation<2, 2> &tria, std::ostream &out) const
       if (y_min_perspective > projection_decomposition[1])
         y_min_perspective = projection_decomposition[1];
 
-      point[0] = cell->vertex(3)[0];
-      point[1] = cell->vertex(3)[1];
+      if (cell->n_vertices() == 4)
```

Comment:
Could you add a comment here why the check is necessary?

Reply:
It's unfortunate that the same code is essentially copied and pasted four times. I would like to clean that up but we shouldn't burden this PR with that refactoring.

Reply:
> Could we avoid writing the extra point?

That's true. @eliasstudiert could you fix this moving the bottom curly bracket down.

Reply:
I've added the comments and fixed writing of the extra point.